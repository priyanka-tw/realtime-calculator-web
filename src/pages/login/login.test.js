import React from 'react';
import {render} from '@testing-library/react';
import 'jest-styled-components';
import WS from "jest-websocket-mock";
import Login from "./login";
import Context from "../../services/global-context-provider/context";

describe('Login page', () => {

    afterEach(() => {
        WS.clean();
    });

    it('should render login page', () => {
        const {container} = render(<Context.Provider value={{setSocket: jest.fn()}}>
            <Login/> </Context.Provider>)

        expect(container).toMatchSnapshot();
    });

    it('should make connection with server over ws protocol', async () => {
        const server = new WS("ws://localhost:8080/ws");
        const mockedFn = jest.fn();
        render(
            <Context.Provider value={{setSocket: mockedFn}}>
                <Login/>
            </Context.Provider>
        )
        await server.connected;

        expect(server.messages).toStrictEqual([]);
        expect(mockedFn).toHaveBeenCalledTimes(1);
    });
});
