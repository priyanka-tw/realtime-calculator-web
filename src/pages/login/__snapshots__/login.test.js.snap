// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page should render login page 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  color: white;
  width: 300px;
  font-size: 16pt;
  font-family: verdana;
  background-color: transparent;
  border: none;
  border-bottom: 2px solid #fff;
  outline: none;
  text-align: center;
  width: 400px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 400px;
  width: 500px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: black;
  margin-top: 100px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  margin-top: 20px;
  width: 200px;
  height: 30px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <input
          class="c3"
          data-testid="username"
          placeholder="Please enter your name"
        />
        <button
          class="c4"
          data-testid="start-button"
        >
          Start
        </button>
      </div>
    </div>
  </div>
   
</div>
`;
